composer/utils/object_store/s3_object_store.py

```diff
@@ -122,19 +122,33 @@ def get_object_size(self, object_name: str) -> int:
             _ensure_not_found_errors_are_wrapped(self.get_uri(object_name), e)
         return obj['ContentLength']
 
-    def upload_object(
-        self,
-        object_name: str,
-        filename: Union[str, pathlib.Path],
-        callback: Optional[Callable[[int, int], None]] = None,
-    ):
+    def upload_object(self,
+                      object_name: str,
+                      filename: Union[str, pathlib.Path],
+                      callback: Optional[Callable[[int, int], None]] = None,
+                      **kwargs):
+        try:
+            from boto3.s3.transfer import S3Transfer
+        except ImportError as e:
+            raise MissingConditionalImportError('streaming', 'boto3') from e
+
         file_size = os.path.getsize(filename)
         cb_wrapper = None if callback is None else lambda bytes_transferred: callback(bytes_transferred, file_size)
+
+        # Validate kwargs
+        if len(kwargs) != 0:
+            if len(kwargs) > 1 or 'ExtraArgs' not in kwargs or not isinstance(kwargs['ExtraArgs'], dict):
+                raise ValueError('S3ObjectStore.upload_object only supports an additional ExtraArgs dictionary.')
```

Comment:
do we need to validate the ExtraArgs bit? can we leave it to the client library so that we don't have to update if they change signature?